okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpChannelBuilder.java

```diff
@@ -144,6 +158,19 @@ public OkHttpChannelBuilder connectionSpec(ConnectionSpec connectionSpec) {
   protected ChannelEssentials buildEssentials() {
     final ExecutorService executor = (transportExecutor == null)
         ? SharedResourceHolder.get(DEFAULT_TRANSPORT_THREAD_POOL) : transportExecutor;
+    switch (negotiationType) {
+      case TLS:
+        if (sslSocketFactory == null) {
+          sslSocketFactory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+        }
+        break;
+      case PLAINTEXT:
+        sslSocketFactory = null;
```

Comment:
> This change all happens on channel builder, the transport's behavior is still "using TLS if we get a SSLSocketFactory".

Oh, I see. Okay. Then only make the change to a local sslSocketFactory variable. Don't change the builder's fields.

> Actually, I call negotiationType(NegotiationType.TLS) inside sslSocketFactory() for the same reason: honor the call even if the user called negotiationType(NegotiationType.PLAINTEXT) previously.

Well, that seems like you overriding, not honoring, their request. If the negotiation type is PLAINTEXT, that was on purpose. Generally the different fields are separate; changing one field doesn't mean others will change. They may be setting the sslSocketFactory _in case_ they end up choosing TLS, but they leave the decision of whether to use TLS to one like that calls `negotiationType()`.

In any case, I can live with the current `sslSocketFactory()` behavior.